examples-ce3/src/main/scala/tofu/example/doobie/TofuDoobieExample.scala

```diff
@@ -0,0 +1,149 @@
+final case class Person(id: Long, name: String, deptId: Long)
+
+@derive(loggable)
+final case class Dept(id: Long, name: String)
+
+// Person SQL algebra
+@derive(representableK, loggingMidTry)
+trait PersonSql[F[_]] {
+  def init: F[Unit]
+  def create(person: Person): F[Unit]
+  def read(id: Long): F[Option[Person]]
+}
+
+object PersonSql extends LoggingCompanion[PersonSql] {
+  def make[DB[_]: Monad: LiftConnectionIO: EmbeddableLogHandler]: PersonSql[DB] = {
+    EmbeddableLogHandler[DB].embedLift(implicit lh => new Impl)
+  }
+
+  final class Impl(implicit lh: LogHandler) extends PersonSql[ConnectionIO] {
+    def init: ConnectionIO[Unit]                     =
+      lsql"create table if not exists person(id int8, name varchar(50), dept_id int8)".update.run.void
+    def create(p: Person): ConnectionIO[Unit]        =
+      lsql"insert into person values(${p.id}, ${p.name}, ${p.deptId})".update.run.void
+    def read(id: Long): ConnectionIO[Option[Person]] =
+      lsql"select id, name, dept_id from person where id = $id"
+        .query[Person]
+        .option
+  }
+}
+
+// Department SQL algebra
+@derive(representableK, loggingMidTry)
+trait DeptSql[F[_]] {
+  def init: F[Unit]
+  def create(dept: Dept): F[Unit]
+  def read(id: Long): F[Option[Dept]]
+}
+
+object DeptSql extends LoggingCompanion[DeptSql] {
+  def make[DB[_]: Monad: LiftConnectionIO: EmbeddableLogHandler]: DeptSql[DB] = {
+    EmbeddableLogHandler[DB].embedLift(implicit lh => new Impl)
+  }
+
+  final class Impl(implicit lh: LogHandler) extends DeptSql[ConnectionIO] {
+    def init: ConnectionIO[Unit]                   =
+      lsql"create table if not exists department(id int8, name varchar(50))".update.run.void
+    def create(d: Dept): ConnectionIO[Unit]        =
+      lsql"insert into department values(${d.id}, ${d.name})".update.run.void
+    def read(id: Long): ConnectionIO[Option[Dept]] =
+      lsql"select id, name from department where id = $id"
+        .query[Dept]
+        .option
+  }
+}
+
+// Storage algebra encapsulates database transactional logic
+@derive(representableK, loggingMidTry)
+trait PersonStorage[F[_]] {
+  def init: F[Unit]
+  def store(person: Person, dept: Dept): F[Unit]
+}
+
+object PersonStorage extends LoggingCompanion[PersonStorage] {
+  def make[F[_]: Apply, DB[_]: Monad: Txr[F, *[_]]](
+      persSql: PersonSql[DB],
+      deptSql: DeptSql[DB]
+  ): PersonStorage[F] = {
+    val impl = new Impl[DB](persSql, deptSql): PersonStorage[DB]
+    val tx   = Txr[F, DB].trans
+    impl.mapK(tx)
+  }
+
+  final class Impl[DB[_]: Monad](persSql: PersonSql[DB], deptSql: DeptSql[DB]) extends PersonStorage[DB] {
+    def init: DB[Unit]                              =
+      deptSql.init >> persSql.init
+    def store(person: Person, dept: Dept): DB[Unit] =
+      deptSql.create(dept) >> persSql.create(person)
+  }
+}
+
+object TofuDoobieExample extends IOApp.Simple {
+
+  val run: IO[Unit] = Dispatcher[IO].use { (disp: Dispatcher[IO]) =>
```

Comment:
could not find implicit value for evidence parameter of type tofu.kernel.types.PerformThrow[[γ$1$]cats.data.Kleisli[cats.effect.IO,tofu.example.doobie.Ctx,γ$1$]]

@Odomontois